dtls.c

```diff
@@ -1841,8 +1841,36 @@ dtls_verify_peer(dtls_context_t *ctx,
 }
 
 #ifdef DTLS_ECC
+/*
+ * Assumes that data_len is at least 1 */
```

Comment:
In fact, this functions assumes `data_len > 1`: After reading `data[0]` in line 1852, there is an access to `data[1]` in line 1856 that could happen with a malformed ASN.1 Integer. I felt more comfortable if the break condition for the while loop in line 1855 had the check for `data_len == 0` as well.

Reply:
I did not put a break condition there as octets could have a value of 0 with a somewhat unusual representation of 0 bytes long.  Have now added
````
+    if (octets && *data_len == 0)
+      return (size_t)-1;
````
and code pushed.